backend/api/src/get-contract-params.ts

```diff
@@ -0,0 +1,180 @@
+import { Bet } from 'common/bet'
+import { getInitialProbability } from 'common/calculate'
+import {
+  BinaryContract,
+  Contract,
+  PseudoNumericContract,
+} from 'common/contract'
+import { HistoryPoint } from 'common/src/chart'
+import {
+  CONTRACT_BET_FILTER,
+  getBets,
+  getTotalBetCount,
+} from 'common/supabase/bets'
+import { getAllComments } from 'common/supabase/comments'
+import {
+  ShareholderStats,
+  getCPMMContractUserContractMetrics,
+  getContractMetricsOutcomeCount,
+  getTopContractMetrics,
+  getTotalContractMetrics,
+} from 'common/supabase/contract-metrics'
+import { getContractFromSlug } from 'common/supabase/contracts'
+import { getUserIsMember } from 'common/supabase/groups'
+import { getRelatedContracts } from 'common/supabase/related-contracts'
+import { removeUndefinedProps } from 'common/util/object'
+import { compressPoints, pointsToBase64 } from 'common/util/og'
+import { createSupabaseClient } from 'shared/supabase/init'
+import { getUser } from 'shared/utils'
+import { z } from 'zod'
+import { APIError, MaybeAuthedEndpoint, validate } from './helpers'
+
+const bodySchema = z.object({
+  contractSlug: z.string(),
+  fromStaticProps: z.boolean(),
+})
+
+export const getcontractparams = MaybeAuthedEndpoint(async (req, auth) => {
+  const { contractSlug, fromStaticProps } = validate(bodySchema, req.body)
+  const db = createSupabaseClient()
+  const contract = await getContractFromSlug(contractSlug, db)
+
+  contractErrors(contract)
+  const groupId =
+    contract.groupLinks && contract.groupLinks.length > 0
+      ? contract.groupLinks[0].groupId
+      : undefined
+  const canAccessContract =
+    // can't access if contract is deleted
+    !contract.deleted &&
+    // can access if contract is not private
+    (contract.visibility != 'private' ||
+      // if contract is private, can't access if in static props
+      (!fromStaticProps &&
+        // otherwise, can access if user can access contract's group
+        auth &&
+        groupId &&
+        (await getUserIsMember(groupId, auth?.uid, db))))
+  if (canAccessContract) {
```

Comment:
Minor style point: if the `!canAccessContract` case is way shorter you might handle that within the if. Then the main body of the function can be the bulk of the code and un-indented.